src/collaborator/client.ts

```diff
@@ -0,0 +1,553 @@
+import { z } from "zod";
+import type {
+  Client,
+  GetInputFunction,
+  RegisterToolsFunction,
+} from "../common/types.js";
+
+export class CollaboratorClient implements Client {
+  name = "Collaborator";
+  prefix = "collaborator";
+  private readonly baseUrl: string;
+  private readonly username: string;
+  private readonly loginTicket: string;
+
+  constructor(baseUrl: string, username: string, loginTicket: string) {
+    this.baseUrl = baseUrl;
+    this.username = username;
+    this.loginTicket = loginTicket;
+  }
+
+  /**
+   * Calls the Collaborator API with the given commands, prepending authentication automatically.
+   * @param commands Array of Collaborator API commands (excluding authentication)
+   * @returns Raw Collaborator API response
+   */
+  async call(commands: any[]): Promise<any> {
+    const url = `${this.baseUrl}/services/json/v1`;
+    // Always prepend authentication command automatically
+    const body = [
+      {
+        command: "SessionService.authenticate",
+        args: { login: this.username, ticket: this.loginTicket },
+      },
+      ...commands,
+    ];
+    const response = await fetch(url, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(body),
+    });
+    if (!response.ok) {
+      throw new Error(
+        `Collaborator API call failed: ${response.status} - ${await response.text()}`,
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * Registers the Collaborator API tool with the MCP server. Accepts commands (excluding authentication).
+   */
+  registerTools(
+    register: RegisterToolsFunction,
+    _getInput: GetInputFunction,
+  ): void {
+    // findReviewById tool
+    register(
+      {
+        title: "Find Collaborator Review By ID",
+        summary: "Finds a review in Collaborator by its review ID.",
+        inputSchema: z.object({
+          reviewId: z.string().describe("The Collaborator review ID to find."),
+        }),
+      },
+      async (args, _extra) => {
+        try {
+          const { reviewId } = args;
+          const commands = [
+            {
+              command: "ReviewService.findReviewById",
+              args: { reviewId },
+            },
+          ];
+          const result = await this.call(commands);
+          return {
+            content: [{ type: "text", text: JSON.stringify(result, null, 2) }],
+          };
+        } catch (err) {
+          return {
+            content: [
+              {
+                type: "text",
+                text: `Error: ${err instanceof Error ? err.message : String(err)}`,
+              },
+            ],
+            isError: true,
+          };
```

Comment:
You don't have to do this – an thrown error will be caught in the `server.ts` and a formatted response returned. This also has the benefit of logging the error to BugSnag.